src/main/java/duke/util/Parser.java

```diff
@@ -0,0 +1,99 @@
+    }
+
+    // extract int from done and delete prompts
+    public static int extractInt(String input) throws DukeException {
+        String[] parts = input.split(" ");//split along the whitespace to get the integer
+        if (parts.length <= 1) {// checking for incomplete prompts
+            throw new DukeException();
+        }
+        String numStr = parts[1];
+        return Integer.valueOf(numStr);
+    }
+
+    public void execute() {
+        String input = sc.nextLine(); // scan the line for the user's input
+
+        boolean shouldContinue = true;
+        while (shouldContinue) {//create loop for the chat
+            try {
+                if (input.startsWith("find ")) {
+                    String desc = input.replaceFirst("find ", "");//remove find from input
+                    if (desc.isBlank()) {//check for incomplete input
+                        throw new DukeException();
+                    }
+                    list.find(desc);
+                    input = sc.nextLine();
+                    continue;
+                } else if (input.startsWith("done ")) {
+                    int numInt = extractInt(input);
+                    list.setIndexDone(numInt);
+                    input = sc.nextLine();
+                    continue;
+                } else if (input.startsWith("delete ")) {
+                    int numInt = extractInt(input);
+                    list.deleteTask(numInt);
+                    input = sc.nextLine();
+                    continue;
+                } else if (input.startsWith("todo ")) {
+                    String task = input.replaceFirst("todo ", "");//remove to-do from input
+                    if (task.isBlank()) {//check for incomplete input
+                        throw new DukeException();
+                    }
+                    list.addToDo(task);
+                    input = sc.nextLine();
+                    continue;
+                } else if (input.startsWith("deadline ")) {
+                    String task = input.replaceFirst("deadline ", "");//remove deadline from input
+                    if (task.isBlank()) {//checking incomplete input
+                        throw new DukeException();
+                    }
+                    String[] parts = task.split("/by ");//split string along keyword /by
+                    list.addDeadline(parts[0], parts[1]);
+                    input = sc.nextLine();
+                    continue;
+                } else if (input.startsWith("event ")) {
+                    String task = input.replaceFirst("event ", "");//remove event from input
+                    if (task.isBlank()) {//checking incomplete input
+                        throw new DukeException();
+                    }
+                    String[] parts = task.split("/at");//split string along keyword /at
+                    list.addEvent(parts[0], parts[1]);
+                    input = sc.nextLine();
+                    continue;
+                }
```

Comment:
Hopefully I am not misunderstanding your code (and correct me if I am!), but I believe that instead of the repeated use of
```java
input = sc.nextLine();
continue;
```

You could merely declare `String input = null;` or `String input;` at line 27, and then place the line 
`input = sc.nextLine()` at the start of the while loop between line 31 and 32. Then you can subsequently remove all lines of `continue` as well. 

Resultantly, in this program it will look at a singular `if` / `if else` block, complete an action, and then loop to get the next line of input.

The `switch` block below can then either be split into a separate method (to shorten this method as mentioned in my other comment), or simply be converted into more `if else` blocks.

Just a suggestion! 😸

Reply:
yea that would be alot cleaner thanks for the suggestion